saleor/core/middleware.py

```diff
@@ -106,3 +113,81 @@ def middleware(request):
         return get_response(request)
 
     return middleware
+
+
+class ReadOnlyMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.blocked_django_url_patterns = [
+            re.compile(r"^/dashboard"),
+            re.compile(r"^/([\w-]+/)?account/$"),
+            re.compile(r"^/([\w-]+/)?account/signup/$"),
+        ]
+        self.allowed_mutations_startswith = ["checkout", "tokenCreate", "tokenVerify"]
```

Comment:
We want:
- `getPaymentToken`
- `createPayment`

Otherwise one won't be able to checkout using dummy payments.